app/src/main/java/org/dhis2/utils/customviews/OptionSetSelectionView.java

```diff
@@ -0,0 +1,236 @@
+    }
+
+    public void setInitValue(String value, List<Option> options, ValueTypeRenderingType renderingType) {
+        this.renderingType = renderingType;
+        this.options = options;
+        this.currentCodeValue = null;
+        if (value != null && value.contains("_os_")) {
+            currentCodeValue = value.split("_os_")[1];
+        } else {
+            currentCodeValue = value;
+        }
+
+        switch (renderingType) {
+            case VERTICAL_CHECKBOXES:
+                checkGroup.setOrientation(LinearLayout.VERTICAL);
+                radioGroup.setVisibility(View.GONE);
+                checkGroup.setVisibility(View.VISIBLE);
+                delete.setVisibility(GONE);
+                setCheckOptions();
+                break;
+            case HORIZONTAL_CHECKBOXES:
+                checkGroup.setOrientation(LinearLayout.HORIZONTAL);
+                radioGroup.setVisibility(View.GONE);
+                checkGroup.setVisibility(View.VISIBLE);
+                delete.setVisibility(GONE);
+                setCheckOptions();
+                break;
+            case VERTICAL_RADIOBUTTONS:
+                radioGroup.setOrientation(LinearLayout.VERTICAL);
+                checkGroup.setVisibility(View.GONE);
+                radioGroup.setVisibility(View.VISIBLE);
+                delete.setVisibility(VISIBLE);
+                setRadioOptions();
+                break;
+            case HORIZONTAL_RADIOBUTTONS:
+                radioGroup.setOrientation(LinearLayout.HORIZONTAL);
+                checkGroup.setVisibility(View.GONE);
+                radioGroup.setVisibility(View.VISIBLE);
+                delete.setVisibility(VISIBLE);
+                setRadioOptions();
+                break;
+        }
+
+        if (delete != null && value != null) {
+            delete.setVisibility(View.VISIBLE);
+        }
+    }
+
+    private void setCheckOptions() {
+        checkGroup.removeAllViews();
+        for (Option option : options) {
+            if (canShowOption(option.uid())) {
+                OptionSetSelectCheckItemBinding optionBinding = OptionSetSelectCheckItemBinding.inflate(inflater, checkGroup, false);
+                optionBinding.setEditable(isEditable);
+                optionBinding.setOptionName(option.displayName());
+                optionBinding.checkBox.setChecked(currentCodeValue != null && (currentCodeValue.equals(option.code()) || currentCodeValue.equals(option.name())));
+                optionBinding.checkBox.setOnCheckedChangeListener((buttonView, isChecked) -> {
+                    if (isChecked) {
+                        listener.onSelectedOption(option.name(), option.code());
+                    } else if (currentCodeValue.equals(option.code())) {
+                        listener.onOptionsClear();
+                    }
+                });
+                checkGroup.addView(optionBinding.getRoot());
+            }
+        }
+        invalidate();
```

Comment:
I think there are some very similar/duplicate code with this method and the below, perhaps we could move things to a common method

Reply:
Same logic, but they are different objects.

Reply:
is there any relationship between those objectS?